// webhookit-feed/lib/webhookit-feed.js
var feedster = require('feedster');



exports.run = function (params, cb) {
    
var items = params.in;
var feed = feedster.createFeed({
    title: params.title,
    description: params.description,
    link: 'http://path/to/blog'
});
// add new item to the feed 
for(var key in items) {
    //console.log(items[key].title);
    feed.addItem({
        title:          items[key].title,
        link:           items[key].url,
        description:    items[key].description,
        pubDate:        items[key].date
    });
}


// generate a RSS string and print to console 
var rss = feed.render();
//console.log(rss);
//console.write(rss);
/*
var feed = new Feed({
    title:          'Feed Title',
    description:    'This is my personnal feed!',
    link:           'http://example.com/',
    image:          'http://example.com/image.png',
    copyright:      'All rights reserved 2013, John Doe',
    
    author: {
        name:       'John Doe',
        email:      'johndoe@example.com',
        link:       'https://example.com/johndoe'
    }
});

for(var key in items) {
    console.log(items[key].title);
    feed.addItem({
        title:          items[key].title,
        link:           items[key].url,
        description:    items[key].description,
        author: [
            {
                name:   'Jane Doe',
                email:  'janedoe@example.com',
                link:   'https://example.com/janedoe'
            },
            {
                name:   'Joe Smith',
                email:  'joesmith@example.com',
                link:   'https://example.com/joesmith'
            }
        ],
        contributor: [
            {
                name:   'Shawn Kemp',
                email:  'shawnkemp@example.com',
                link:   'https://example.com/shawnkemp'
            },
            {
                name:   'Reggie Miller',
                email:  'reggiemiller@example.com',
                link:   'https://example.com/reggiemiller'
            }
        ],
        date:           items[key].date,
        image:          "/"
    });
}
feed.render('rss-2.0');
*/
cb({"out": rss}); 


};



exports.definition = {
    "name": "feed",
    "category": "Templating",
    "container": {
        "icon": "/images/icons/rss_go.png",
        "xtype": "WireIt.FormContainer",
        "title": "Output RSS",
        "fields": [
            {
                "type": "string",
                "name": "title",
                "wirable": false,
                label: "RSS Title"
            },
            {
                "type": "string",
                "name": "description",
                "wirable": false,
                label: "RSS Description"
            }

        ],
        "terminals": [
            {"name": "in", "direction": [0, -1], "offsetPosition": [75, -15], "ddConfig": {
                    "type": "input",
                    "allowedTypes": ["output"]
                },
                "nMaxWires": 1
            },
            {"name": "out", "direction": [0, 1], "offsetPosition": {"left": 75, "bottom": -15}, "ddConfig": {
                    "type": "output",
                    "allowedTypes": ["input"]
                }
            }
        ]
    }
};
