// webhookit-xpath/lib/webhookit-xpath.js
var xpath = require('xpath')
      , dom = require('xmldom').DOMParser;




exports.run = function(params, cb) {
var doc = new dom().parseFromString(params["url"]);
    var nodes = xpath.select('//*[@id="maincontent"]/div/ul/li', doc, false);
    //console.log(nodes[0].localName + ": " + nodes[0].firstChild.data)
    //console.log("node: " + nodes[0].toString())
    console.log(nodes[1]);
    //for (i = 0; i < nodes.length; i++) {

    //cb({out: nodes[i].localName + ": " + nodes[0].data});
    //}
    
    


};


exports.definition = {
    "name": "xpath",
    "category": "Operators",
    "container": {
        "icon": "/images/icons/connect.png",
        "xtype": "WireIt.FormContainer",
        "title": "xpath",
        "fields": [
        {
            "type": "string",
            "name": "url",
            "wirable": true,
            label: "Url"
        },
        {
            "type": "select",
            "name": "method",
            "wirable": false,
            label: "Method",
            choices: ["fetchPage", "scrape", "spider"]
        },
        {
            "type": "select",
            label: "encoding",
            "name": "encoding",
            choices: ["application/x-www-form-urlencoded", "application/json", "application/xml"]
        },
        {
            "type": "list",
            "name": "urlparams",
            listLabel: 'Parameters',
            elementType: {
                type: 'combine',
                fields: [{typeInvite: 'field'},{ typeInvite: 'value',wirable: true}],
                separators: [false, ":", false]
            }
        }
        ],
        "terminals": [
        {
            "name": "out",
            "direction": [0, 1],
            "offsetPosition": {
                "left": 86,
                "bottom": -15
            },
            "ddConfig": {
                "type": "output",
                "allowedTypes": ["input"]
            }
        }
        ]
    }
};
